tests/functional/test_metrics.py

```diff
@@ -320,8 +320,8 @@ def test_schema_no_metrics(service, namespace):
 
 
 def test_exceed_number_of_dimensions(metric, namespace):
-    # GIVEN we have more dimensions than CloudWatch supports
-    dimensions = [{"name": f"test_{i}", "value": "test"} for i in range(11)]
+    # GIVEN we have more dimensions than CloudWatch supports (N+1)
+    dimensions = [{"name": f"test_{i}", "value": "test"} for i in range(31)]
```

Comment:
perhaps, let's start using the constant `MAX_DIMENSIONS` so these can be updated/failed accordingly?